### PR TITLE
rpc: add wrapper for listtransactions command with include_watchonly parameter

### PR DESCRIPTION
**Pull request by mycroft**

### CODE REVIEW

File: rpcclient/wallet.go [hunk 1]
```diff
@@ -141,6 +141,24 @@ func (c *Client) ListTransactionsCountFrom(account string, count, from int) ([]b
...
+// to get the result of the RPC at some future time by invoking the Receive
+// function on the returned instance.
+//
+// See ListTransactionsCountFrom for the blocking version and more details.
+func (c *Client) ListTransactionsCountFromWatchAsync(account string, count, from int, watch bool) FutureListTransactionsResult {
+	cmd := btcjson.NewListTransactionsCmd(&account, &count, &from, &watch)
+	return c.sendCmd(cmd)
+}
+
+// ListTransactionsCountFrom returns a list of the most recent transactions up
```
Comment (torkelrogstad) [inline rpcclient/wallet.go]: This doc comment needs to be tweaked, should start with `ListTransactionsCountFromWatch` and mention that it includes watch-only wallet TXs

**Comment by mycroft** [comment]
Hello @torkelrogstad - I've updated comments as requested. Thanks!

**Comment by jakesyl** [comment]
@jcvernaleo (as per #1530)

- low priority
- enhancement

Mergeable

**Comment by mycroft** [comment]
Hello, I've squashed the commits as requested & rebased it (no impact).

Thanks!

**Comment by jcvernaleo** [comment]
Done with #1628

**Pull request closed without merging**